File: algorithms/1_easy/M/01974_minimum_time_to_type_word_using_special_typewriter/solution.py
class Solution:
    def minTimeToType(self, word: str) -> int:
        circle_pos = 0
        cost = 0

        for ch in word:
            new_pos = ord(ch) - ord('a')
            curr_diff = abs(new_pos - circle_pos)

            min_cost = min(curr_diff, 26 - curr_diff)

            cost += min_cost + 1
            circle_pos = new_pos

        return cost

File: algorithms/1_easy/M/01974_minimum_time_to_type_word_using_special_typewriter/solution.cpp
class Solution {
public:
    int minTimeToType(string word) {
        int circle_pos{0};
        int cost{0};

        for (const char& ch : word) {
            int new_pos{static_cast<int>(ch - 'a')};
            int curr_diff{abs(new_pos - circle_pos)};

            int min_cost = std::min(curr_diff, 26 - curr_diff);

            cost += min_cost + 1;
            circle_pos = new_pos;
        }

        return cost;
    }
};
